src/station_cmd.cpp

```diff
@@ -4529,6 +4657,12 @@ CommandCost ClearTile_Station(TileIndex tile, DoCommandFlag flags)
 				if (remove_road.Failed()) return remove_road;
 			}
 			return RemoveRoadStop(tile, flags);
+		case STATION_ROADWAYPOINT:
+			if (IsDriveThroughStopTile(tile)) {
+				CommandCost remove_road = CanRemoveRoadWithStop(tile, flags);
+				if (remove_road.Failed()) return remove_road;
+			}
```

Comment:
I think this is a copy-paste mistake. Why would road waypoints also be drive through stops?

Reply:
Yes, the condition is redundant here as all road waypoints are drive-through. (Bay waypoints would be a bit useless).
I'll fix this later,